scrape/scrape.go

```diff
@@ -662,19 +667,26 @@ mainLoop:
 				time.Since(last).Seconds(),
 			)
 		}
-		b := sl.buffers.Get(sl.lastScrapeSize)
-		buf := bytes.NewBuffer(b)
+		temp, err := sl.buffers.Get(sl.buffers, sl.lastScrapeSize)
```

Comment:
I'm not sure I quite understand this. Why would we pass the object itself as the first argument to a method on itself?